app/models/event.rb

```diff
@@ -119,4 +119,8 @@ def created_at
   def _location_
     @object.location
   end
+
+  def as_json
+    {location: _location_.as_json, agent: agent.as_json, payload: @payload.as_json, created_at: created_at.as_json}
```

Comment:
It was, which I agree shouldn't be an issue. However in that case we should not call the local [agent](https://github.com/cantino/huginn/pull/1716/files/19c48b1ef5f9df79a2d816fcdd6bba50f924ea85#diff-6ecd61439b970b0e5ae49be738861446L107) method but return the agent even if `agent` is a key in the payload.  To make it consistent with the way an Agent is available in Liquid I added the `to_liquid` call before transforming the Agent to a safe object.